src/components/Header/NotificationMenu.tsx

```diff
@@ -0,0 +1,217 @@
+import {
+  Avatar,
+  AvatarBadge,
+  Flex,
+  MenuButtonProps,
+  MenuListProps,
+  Text,
+} from "@chakra-ui/react"
+import { Menu, Spinner } from "@opengovsg/design-system-react"
+import { ContextMenuItem } from "components/ContextMenu/ContextMenuItem"
+import { PropsWithChildren, useEffect, useState } from "react"
+import { BiBell, BiDownArrowAlt } from "react-icons/bi"
+
+import {
+  useGetAllNotifications,
+  useGetNotifications,
+  useUpdateReadNotifications,
+} from "hooks/notificationHooks"
+import useRedirectHook from "hooks/useRedirectHook"
+
+import { convertDateToTimeDiff } from "utils/dateUtils"
+import { getAvatarName, getNotificationIcon } from "utils/notificationUtils"
+
+const NotificationMenuButton = (props: MenuButtonProps): JSX.Element => {
+  return (
+    <Menu.Button
+      variant="clear"
+      px="0"
+      iconSpacing="0.5rem"
+      rightIcon={undefined}
+      // color="secondary.300"
+      _focus={{}}
+      {...props}
+    />
+  )
+}
+
+interface NotificationMenuItemProps {
+  name?: string
+  icon?: JSX.Element
+  link: string
+  time: string
+  isRead: boolean
+}
+
+export const NotificationMenuItem = ({
+  name,
+  icon,
+  link,
+  time,
+  isRead,
+  children,
+}: PropsWithChildren<NotificationMenuItemProps>): JSX.Element => {
+  const { setRedirectToPage } = useRedirectHook()
+  return (
+    <ContextMenuItem
+      backgroundColor={isRead ? "background.default" : "blue.50"}
+      onClick={() => setRedirectToPage(link)}
+    >
+      <Flex w="100%">
+        <Avatar icon={icon || <></>} name={name ? getAvatarName(name) : ""} />
+        <Text
+          w="100%"
+          pl="1rem"
+          noOfLines={3}
+          fontSize="0.75rem"
+          color="text.description"
+          title={typeof children === "string" ? children : undefined}
+        >
+          {children}
+        </Text>
+        <Text
+          pl="0.5rem"
+          alignSelf="start"
+          color="text.helper"
+          fontSize="0.75rem"
+        >
+          {time}
+        </Text>
+      </Flex>
+    </ContextMenuItem>
+  )
+}
+
+interface NotificationMenuProps {
+  siteName: string
+  menuListProps?: MenuListProps
+}
+
+export const NotificationMenu = ({
+  siteName,
+  menuListProps,
+}: NotificationMenuProps): JSX.Element => {
+  const [displayAll, setDisplayAll] = useState(false)
+  const [hasNotification, setHasNotification] = useState(false)
+  const {
+    data: allNotificationData,
+    refetch: refetchAllNotificationData,
+    isLoading: isGetAllNotificationLoading,
+  } = useGetAllNotifications(siteName)
+  const {
+    data: recentNotificationData,
+    refetch: refetchRecentNotificationData,
+  } = useGetNotifications(siteName)
```

Comment:
I opted for this separation because we're going to be frequently querying for recent notifications (as we need this information on every page to determine if there are any new notifications), so I wanted to make it lightweight, while fetching all notifications can be slower because it's a user triggered operation (when clicking to see more notifications)

The design doc talks about it as one of the eng decisions I made! https://www.notion.so/opengov/Comments-Notifications-4332401f397c487e9a6e3bf641b23814

Reply:
Ok! Thanks for the clarification!